### PR TITLE
Logger redesign

### PR DESCRIPTION
**Pull request by ovstinga**
This **PR** mostly targets the wasm logger redesign.

There are still some things that remain to be addressed:  

- [x] move set_log_level call into its own capi call

### CODE REVIEW

File: exec-service-wasmer/src/wasmer_executor.rs [hunk 1]
```diff
@@ -1,101 +1,62 @@
...
 
 impl WasmerExecutorData {
-    pub(crate) fn debug(&self, message: &str) {
-        if self.log_level >= WasmerExecutorLogLevel::Debug {
-            println!("[DEBUG] {message}");
+    fn set_vm_hooks_ptr(&mut self, vm_hooks_ptr: *mut c_void) -> Result<(), ExecutorError> {
+        if let Some(vm_hooks) = Rc::get_mut(&mut self.vm_hooks) {
+            vm_hooks.set_vm_hooks_ptr(vm_hooks_ptr);
+            return Ok(());
         }
```
Comment (andrei-marinica) [inline exec-service-wasmer/src/wasmer_executor.rs:22]: you can use else and not need to `return`

File: exec-service-wasmer/src/wasmer_executor.rs [hunk 2]
```diff
@@ -1,101 +1,62 @@
...
-        }
-
-        Err(Box::new(ServiceError::new(
-            "WasmerExecutor opcodes cost configuration error",
-        )))
+        info!("Setting opcode cost...");
+        self.data.borrow_mut().set_opcode_cost(opcode_cost)
     }
 
     fn set_execution_log_level(&mut self, value: u64) -> Result<(), ExecutorError> {
```
Comment (andrei-marinica) [inline exec-service-wasmer/src/wasmer_executor.rs:50]: We should move this to the service, or CAPI

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

File: exec-service-wasmer/src/wasmer_executor.rs (hunk 2 above)
Comment (ovstinga) [reply exec-service-wasmer/src/wasmer_executor.rs:50]: Right, already noted this down in description as todo

**Review by andrei-marinica — APPROVED** [review]

File: exec-service-wasmer/src/wasmer_executor.rs (hunk 1 above)
Comment (ovstinga) [reply exec-service-wasmer/src/wasmer_executor.rs:25]: Done

File: exec-service-wasmer/src/wasmer_executor.rs (hunk 2 above)
Comment (ovstinga) [reply exec-service-wasmer/src/wasmer_executor.rs]: Done

**Review by camilbancioiu — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**